apps/full-site-editing/full-site-editing-plugin/donations/edit.js

```diff
@@ -11,34 +11,45 @@ import Donations from './donations';
 import LoadingError from './loading-error';
 import LoadingStatus from './loading-status';
 import UpgradePlan from './upgrade-plan';
+import fetchDefaultProducts from './fetch-default-products';
 import fetchStatus from './fetch-status';
 
 const Edit = () => {
 	const [ isLoading, setIsLoading ] = useState( true );
 	const [ loadingError, setLoadingError ] = useState( '' );
 	const [ shouldUpgrade, setShouldUpgrade ] = useState( false );
 	const [ stripeConnectUrl, setStripeConnectUrl ] = useState( false );
+	const [ products, setProducts ] = useState( [] );
 	const [ upgradeUrl, setUpgradeUrl ] = useState( '' );
 
+	const apiError = ( message ) => {
+		setLoadingError( message );
+		setIsLoading( false );
+	};
+
 	const mapStatusToState = ( result ) => {
 		if ( ( ! result && typeof result !== 'object' ) || result.errors ) {
 			setLoadingError( __( 'Could not load data from WordPress.com.', 'full-site-editing' ) );
 		} else {
 			setShouldUpgrade( result.should_upgrade_to_access_memberships );
 			setUpgradeUrl( result.upgrade_url );
 			setStripeConnectUrl( result.connect_url );
-		}
 
-		setIsLoading( false );
```

Comment:
I see this has been copied to `apiError` likely to handle the async response of `fetchDefaultProducts ` but I think we should run this after setting the products as well to mimic the previous behavior.

Reply:
Doesn't L48 (last line of `mapStatusToState`) handle that?

Reply:
oh right, I missed it in the review